src/Neo/SmartContract/Helper.cs

```diff
@@ -360,7 +360,15 @@ internal static bool VerifyWitness(this IVerifiable verifiable, ProtocolSettings
                 engine.LoadScript(invocationScript, configureState: p => p.CallFlags = CallFlags.None);
 
                 if (engine.Execute() == VMState.FAULT) return false;
-                if (!engine.ResultStack.Peek().GetBoolean()) return false;
+                if (engine.ResultStack.Count == 0) return false;
+                try
+                {
+                    if (!engine.ResultStack.Peek().GetBoolean()) return false;
```

Comment:
It's still incorrect for the case of multiple values returned. There MUST be exactly one result.

Reply:
done